test/test_wiki_trac_rst_convert.py

```diff
@@ -107,6 +107,168 @@ def test_several_trac_wiki_rst_links_with_content(self):
+            '`Buildbot <https://chevah.com/buildbot/>`_\n',
+            '[https://chevah.com/buildbot/ Buildbot]'
+        )
+
+    def test_tracwiki_wiki_link(self):
+        """
+        Process wiki links from TracWiki format to GitHub-compatible
+        RST wiki links.
+
+        There are various combinations of:
+        * link text different from article name
+        * link text the same as article name, and
+        * no link text, only article name
+        """
+        self.assertConvertedContent(
+            '`Project management and administration <Administrative>`_\n',
+            '[wiki:Administrative Project management and administration]'
+        )
+        self.assertConvertedContent(
+            '`<Administrative>`_\n',
+            '[wiki:Administrative Administrative]'
+        )
+        self.assertConvertedContent(
+            '`<Administrative>`_\n',
+            '[wiki:"Administrative"]'
+        )
+        self.assertConvertedContent(
+            '`<Administrative-AllHandMeeting-Past>`_\n',
+            '[wiki:"Administrative/AllHandMeeting/Past"]'
+        )
+        self.assertConvertedContent(
+            '`<Infrastructure-Services-FileServer>`_\n',
+            '`[wiki:Infrastructure/Services/FileServer]`:trac:'
+        )
+        self.assertConvertedContent(
+            '`Overton <Infrastructure-Machines-Overton>`_\n',
+            '`[wiki:Infrastructure/Machines/Overton Overton]`:trac:'
+        )
+
+    def test_trac_ticket(self):
+        """
+        Trac ticket references are converted to a hyperlink.
+        This use case requires `config.py` with the following setting:
+
+        TRAC_TICKET_PREFIX = 'https://trac.chevah.com/ticket/'
+        """
+        self.assertConvertedContent(
+            '`Trac #738 <https://trac.chevah.com/ticket/738>`_\n',
+            ':trac:`#738`'
+        )
+
+    def test_heading(self):
+        """
+        Converts headings to RST, which have an equal-sign-underline.
+        Also handles the multiline case.
+
+        Headings in TracWiki have single equal signs around them.
+        """
+        self.assertConvertedContent(
+            'Heading\n=======\n',
+            '= Heading ='
+        )
+        self.assertConvertedContent(
+            'Policy and Process\n'
+            '==================\n\n'
+            'Some text\n',
+
+            '= Policy and Process =\n\n'
+            'Some text'
+        )
+
+    def test_subheading(self):
+        """
+        Converts subheadings to RST, which have a dash-underline.
+        Also handle the multiline case.
+
+        Subheadings in TracWiki have double equal signs around them.
+        """
+        self.assertConvertedContent(
+            'Subheading\n----------\n',
+            '== Subheading =='
+        )
+        self.assertConvertedContent(
+            'Subheading\n----------\n\n'
+            'Some text\n',
+
+            '== Subheading ==\n\n'
+            'Some text'
```

Comment:
Just a minor test that is missing.

Conversion from source

```
== Subheading ==
Test right below the heading
```


It would be nice to convert it as something like this with one extra newline added...

```
Subheading
----------

Test right below the heading
```

but this conversion is also ok


```
Subheading
----------
Test right below the heading
```

no need to spend more time here

Reply:
The test right below this one (with "Some text") ensures two `\n`s already. But I will format it like [you suggested for the Heading test](https://github.com/chevah/trac-to-github/pull/6#discussion_r565509991), since it is clear that the `\n`s on the same line are confusing :)

I wondered what `black` does. It's a problem, but they are [aware of it](https://github.com/psf/black/issues/330). Look what it made of the two tests:

```python
self.assertConvertedContent("Subheading\n" "----------", "== Subheading ==")
self.assertConvertedContent(
    "Subheading\n" "----------\n" "\n" "Some text",
    "== Subheading ==\n" "\n" "Some text",
)
```